src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -1895,12 +1967,63 @@ void ShenandoahHeap::set_gc_state(uint mask, bool value) {
   assert(ShenandoahSafepoint::is_at_shenandoah_safepoint(), "Must be at Shenandoah safepoint");
   _gc_state.set_cond(mask, value);
   _gc_state_changed = true;
+  // Check that if concurrent weak root is set then active_gen isn't null
+  assert(!is_concurrent_weak_root_in_progress() || active_generation() != nullptr, "Error");
+  shenandoah_assert_generations_reconciled();
```

Comment:
Why all of this is checked here? I would have thought `gc_state` machinery should only check things related to gc-state.

Reply:
Likely debugging detritus that I unintentionally left behind. Feel free to delete.
I think the `shenandoah_assert_generations_reconciled()` is already checked in a most spots where it's needed.